hypothesis/lms
# Raise the right exception to correctly report auth issues

For LTI 1.1 raise lms.ValidationError instead of marshmallow's one which results in a 403 instead of a unhandled 500.

For LTI 1.3 correctly pass the `messages` argument.


## Testing 

- Make sure you have `make devdata` up to data


- Apply a diff like 


```diff
diff --git a/lms/validation/authentication/_lti.py b/lms/validation/authentication/_lti.py
index 227d9ad7..5e99ac8e 100644
--- a/lms/validation/authentication/_lti.py
+++ b/lms/validation/authentication/_lti.py
@@ -52,7 +52,7 @@ class LTI11AuthSchema(LTIV11CoreSchema):
         try:
             application_instance = (
                 self._application_instance_service.get_by_consumer_key(
-                    kwargs["oauth_consumer_key"]
+                    kwargs["oauth_consumer_key"] + "NOPE"
                 )
             )
         except ApplicationInstanceNotFound as err:
@@ -116,7 +116,7 @@ class LTI13AuthSchema(LTIV11CoreSchema):
         try:
             application_instance = (
                 self._application_instance_service.get_by_deployment_id(
-                    kwargs["iss"], kwargs["aud"], kwargs["deployment_id"]
+                    kwargs["iss"], kwargs["aud"], kwargs["deployment_id"] + "NOPE"
                 )
             )
         except ApplicationInstanceNotFound as err:
```


On `main`:

- LTI 1.1 https://hypothesis.instructure.com/courses/125/assignments/873 

Gives a 500

- LTI 1.3 https://hypothesis.instructure.com/courses/319/assignments/3308 
 
-  `You're not authorized to view this page`



Switch to `lti-validation-error`, you'll get a better error in both assignments.
